src/theory/smt_engine_subsolver.cpp

```diff
@@ -42,40 +69,37 @@ Result quickCheck(Node& query)
 }
 
 void initializeSubsolver(std::unique_ptr<SolverEngine>& smte,
-                         const Options& opts,
-                         const LogicInfo& logicInfo,
+                         const SubsolverSetupInfo& info,
                          bool needsTimeout,
                          unsigned long timeout)
 {
   NodeManager* nm = NodeManager::currentNM();
-  smte.reset(new SolverEngine(nm, &opts));
+  smte.reset(new SolverEngine(nm, &info.d_opts));
   smte->setIsInternalSubsolver();
-  smte->setLogic(logicInfo);
+  smte->setLogic(info.d_logicInfo);
   // set the options
   if (needsTimeout)
   {
     smte->setTimeLimit(timeout);
   }
+  // set up separation logic heap if necessary
+  if (!info.d_sepLocType.isNull() && !info.d_sepDataType.isNull())
+  {
+    smte->declareSepHeap(info.d_sepLocType, info.d_sepDataType);
+  }
 }
 void initializeSubsolver(std::unique_ptr<SolverEngine>& smte,
                          const Env& env,
                          bool needsTimeout,
                          unsigned long timeout)
 {
-  initializeSubsolver(
-      smte, env.getOptions(), env.getLogicInfo(), needsTimeout, timeout);
-  // set up separation logic heap if necessary
-  TypeNode sepLocType, sepDataType;
-  if (env.getSepHeapTypes(sepLocType, sepDataType))
-  {
-    smte->declareSepHeap(sepLocType, sepDataType);
-  }
+  SubsolverSetupInfo ssi(env);
+  initializeSubsolver(smte, ssi, needsTimeout, timeout);
```

Comment:
This overload should not be necessary. `initializeSubsolver(..., d_env, ...)` should work as expected due to implicit construction of the `SubsolverSetupInfo` instance.

Reply:
I'd prefer to have this be written explicitly.  I don't like the idea of having explicit construction happening in this case.